compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -31,15 +32,28 @@ fn parse_args<'a>(
     is_global_asm: bool,
 ) -> Result<AsmArgs, DiagnosticBuilder<'a>> {
     let mut p = ecx.new_parser_from_tts(tts);
+    let sess = &ecx.sess.parse_sess;
+    parse_asm_args(&mut p, sess, sp, is_global_asm)
+}
+
+// Primarily public for rustfmt consumption.
+// Internal consumers should continue to leverage `expand_asm`/`expand__global_asm`
+pub fn parse_asm_args<'a>(
+    p: &mut Parser<'a>,
+    sess: &'a ParseSess,
```

Comment:
Realize that it's just the handler that's currently needed for the diagnostics (source map wasn't actually needed for the initial parsing) but figured it might be easier to just pass a ref of the session anyway so that any future changes needing more from the sess could be made without requiring signature and call site updates.

Lmk if it'd be better to reduce this to just the handler though